server/player/player.go

```diff
@@ -557,9 +557,38 @@ func (p *Player) Hurt(dmg float64, source damage.Source) (float64, bool) {
 		p.Exhaust(0.1)
 
 		damageToArmour := int(math.Max(math.Floor(dmg/4), 1))
+		var damageToAttacker int
+		thornsArmour := make([]item.Stack, 0, 4)
 		for slot, it := range p.armour.Slots() {
 			if _, ok := it.Item().(item.Durable); ok {
-				_ = p.armour.Inventory().SetItem(slot, p.damageItem(it, damageToArmour))
+				var thornsDamage int
+				if e, ok := it.Enchantment(enchantment.Thorns{}); ok && rand.Intn(99) < e.Level()*15 {
+					thornsArmour = append(thornsArmour, it)
+					thornsDamage = 1
+					if e.Level() > 10 {
+						damageToAttacker += e.Level() - 10
+					} else {
+						damageToAttacker += 1 + rand.Intn(3)
```

Comment:
```suggestion
						damageToAttacker += 1 + rand.Intn(4)
```

should be 1-4 damage

Reply:
Wouldn't this make it 1-5?